src/Core/Checkout/Promotion/Gateway/PromotionGateway.php

```diff
@@ -25,7 +29,9 @@ public function __construct(private readonly EntityRepository $promotionReposito
      * Gets a list of promotions for the provided criteria and
      * sales channel context.
      *
-     * @return EntityCollection<PromotionEntity>
+     * @deprecated tag:v6.7.0 - Return type will be natively changed to PromotionCollection
```

Comment:
```suggestion
     * @deprecated tag:v6.7.0 - reason:return-type-change - Return type will be changed to PromotionCollection
```

with this PHPStan will not complain

Reply:
Done, thank you :-)